core/arch/interp.c

```diff
@@ -5771,6 +5771,189 @@ mangle_x64_ib_in_trace(dcontext_t *dcontext, instrlist_t *trace, instr_t *target
 }
 #endif
 
+#ifdef AARCH64
+/* Prior to indirect branch trace mangling, we check previous byte blocks
+ * before each indirect branch if they were mangled properly.
+ * An indirect branch:
+ *    br jump_target_reg
+ * is mangled into (see mangle_indirect_jump in mangle.c):
+ *    str IBL_TARGET_REG, TLS_REG2_SLOT
+ *    mov IBL_TARGET_REG, jump_target_reg
+ *    b ibl_routine or indirect_stub
+ * This function is used by mangle_indirect_branch_in_trace;
+ * it removes the two mangled instructions
+ * and returns the jump_target_reg id.
+ */
+#    define STR_IMM_ENC 0xf9000000
+#    define SAVE_IBL_REG_ENC                                                       \
+        ((STR_IMM_ENC | 16 >> 3 << 10) | (dr_reg_stolen - DR_REG_START_GPR) << 5 | \
+         (IBL_TARGET_REG - DR_REG_START_GPR))
+#    define MOV_IBL_REG_ENC_MASK 0xff00ffff
+#    define MOV_IBL_REG_MASK 0x001f0000
+#    define MOV_IBL_REG_ENC (0xaa0003e0 | (IBL_TARGET_REG - DR_REG_START_GPR))
+
+static reg_id_t
+check_and_remove_patched_ibl(dcontext_t *dcontext, instrlist_t *trace, instr_t *targeter)
```

Comment:
`dcontext` seems unused here. Do we need it?

Reply:
Nope, will remove it